dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/WebModuleOpenAPIProvider.java

```diff
@@ -56,12 +75,68 @@ public String getOpenAPIDocument(Format format) {
         return document;
     }
 
+    /**
+     * @see io.openliberty.microprofile.openapi20.OpenAPIProvider#getOpenAPIDocument(List<Server>, Format)
+     */
+    @Override
+    public String getOpenAPIDocument(List<Server> servers, Format format) {
+        // Create the variable to return
+        String document = null;
+
+        // Make sure that the list of servers is valid
+        if (servers != null && !servers.isEmpty()) {
+            /*
+             * We only store a single copy of the OpenAPI model for a given application, but the OpenAPI document could
+             * be requested by multiple clients concurrently.  As a result, we need to synchronize operations that
+             * update the model to ensure consistent behaviour. 
+             */
+            synchronized (openAPIModel) {
+                // Set the servers in the OpenAPI model and generate the OpenAPI document
+                openAPIModel.setServers(servers);
+                document = OpenAPIUtils.getOpenAPIDocument(openAPIModel, format);
+            }
```

Comment:
This is much better.

However, is there a possibility that anything else could be reading `openAPIModel` at the same point that this modification is happening? (E.g. line 125)

I suspect that you need to synchronize around every access to `openAPIModel` to be safe here.